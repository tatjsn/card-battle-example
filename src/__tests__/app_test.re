open Jest;
open ExpectJs;

let cards: list Card.t = [
  { name: "Card1", attacks: [Buster] },
];

let _ =

describe "src/app.re" (fun _ => {

  test "is defined" (fun _ => {
    let component = ReactShallowRenderer.renderWithRenderer <App cards=cards />;
    expect (Js.Undefined.return component) |> toBeDefined;
  });

  test "renders" (fun _ => {
    let tree = App.createElement cards::cards children::[] ()
      |> ReactShallowRenderer.renderWithRenderer;

    expect tree |> toMatchSnapshot;
  });

});
