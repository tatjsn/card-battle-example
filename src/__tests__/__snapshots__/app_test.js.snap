// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app.re renders 1`] = `
<div>
  <Prepare
    reasonProps={
      Array [
        Array [
          Array [
            "Card1",
            Array [
              2,
              0,
            ],
          ],
          0,
        ],
        0,
        [Function],
      ]
    }
  />
</div>
`;
